src/server/cluster/cluster_family.cc

```diff
@@ -689,6 +691,37 @@ void ClusterFamily::DflySlotMigrationStatus(CmdArgList args, ConnectionContext*
   return rb->SendSimpleString(state_to_str(MigrationState::C_NO_STATE));
 }
 
+void ClusterFamily::DflyClusterMigrationFinalize(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  auto sync_id = parser.Next<uint32_t>();
+
+  if (auto err = parser.Error(); err) {
+    return cntx->SendError(err->MakeReply());
+  }
+
+  auto migration_it = outgoing_migration_jobs_.find(sync_id);
+  if (migration_it == outgoing_migration_jobs_.end()) {
+    return cntx->SendError(absl::StrCat("finalize invalid session id ", sync_id));
+  }
+
+  shard_set->pool()->AwaitFiberOnAll([migration_it](auto*) {
+    if (const auto* shard = EngineShard::tlocal(); shard)
+      migration_it->second->Finalize(shard->shard_id());
```

Comment:
make sure Finalize is done only after reaching stable state. you can add a todo for it if you dont what to implement it right now